shell/common/display_manager.cc

```diff
@@ -47,4 +52,22 @@ void DisplayManager::CheckDisplayConfiguration(
   }
 }
 
+void DisplayManager::UpdateRefreshRateIfNecessary(
+    DisplayUpdateType update_type,
+    fml::TimePoint vsync_start_time,
+    fml::TimePoint frame_target_time) {
+  auto is_frame_rate_same = [](double fr1, double fr2) {
+    const double kRefreshRateCompareEpsilon = 1;
+    return fabs(fr1 - fr2) < kRefreshRateCompareEpsilon;
+  };
+
+  double new_frame_rate =
+      round((1 / (frame_target_time - vsync_start_time).ToSecondsF()));
+  if (!is_frame_rate_same(GetMainDisplayRefreshRate(), new_frame_rate)) {
```

Comment:
On Android now this should always return false, but I'm curious if iOS can do something closer to what Android does and just supply a display object that always returns the right value for `GetMainDisplayRefreshRate` rather than polling/checking every frame.

Reply:
On iOS, we can only get the frame time from `target_time - start_time`. We can have a platform(iOS) layer object to manage that calculation so it is abstracted from other platforms.

I think it is true for all the platforms that `target_time - start_time` represents the frame duration, and doing such calculation per frame is not expensive?

Reply:
How do you calculate the target_time on iOS?

On Android, it's calculated using the frame rate - requesting the frame rate every frame is expensive on Android in some vendor implementations though.